src/core/Job.ts

```diff
@@ -166,42 +173,67 @@ export default class Job {
                     return filteredLocations.length > 0;
                 });
 
+            // Job post region is not in our list
+            const isRegionUnknown = !this.config.regionNames.find((location) => {
+                return post.location.match(new RegExp(location, "i"))
+            });                
+
             const nameCheck = !similarPost || similarPost.name === post.name;
-            if (!isProtected && locationCheck && nameCheck)
-                postToDelete.push(post);
+            if (!isProtected && nameCheck) {
+                if (locationCheck) {
+                    postToDelete.push(post);
+                } else if (isRegionUnknown) {
+                    postsUnknownRegion.push(post)
+                }
+            }
+                
+            
         }
-        return postToDelete;
+        return [postToDelete, postsUnknownRegion];
     }
 
     public async deletePosts(
-        config: Config,
         jobData: JobInfo,
         enteredRegions?: string[],
         similarPostID?: number
     ) {
         this.spinner.start(`Starting to delete job posts.`);
         const posts: PostInfo[] = jobData.posts;
 
-        const postToDelete = this.filterPostsToDelete(
+        const [postToDelete, postsUnknownRegion] = this.filterPostsToDelete(
             posts,
             similarPostID,
             enteredRegions
         );
 
-        // We need some ids to be able to delete posts.
-        // They are the same for all boards.
+        // We need `publishStatusId` and `unpublishStatusId`
+        // to be able to delete posts. They are the same for all boards.
         const boardToPost = await this.getBoardToPost();
 
-        const referrer = joinURL(
-            config.greenhouseUrl,
-            `/plans/${jobData.id}/jobapp`
-        );
+        await this.deleteJobPosts(postToDelete, boardToPost, jobData);
+        
+        // Check if there are posts on unrecognised regions and prompt to delete them
+        if (postsUnknownRegion.length > 0) {
+            const prompt = new Toggle({
+                message: `${postsUnknownRegion.length} job posts on unrecognised regions. Do you want to delete them?`,
```

Comment:
```suggestion
                message: `${postsUnknownRegion.length} job posts in unrecognised regions. Do you want to delete them?`,
```

Reply:
Ops, last minute change. Should've followed my first intuition :smiley: